pkg/operator/ceph/cluster/osd/osd.go

```diff
@@ -380,186 +243,46 @@ func (c *Cluster) getExistingOSDDeploymentsOnPVCs() (map[string]*apps.Deployment
-			c.updateOSDStatus(n.Name, status)
-		}
-	}
-
-	logger.Infof("start osds after provisioning is completed, if needed")
-	c.completeProvision(config)
-}
-
-func (c *Cluster) runJob(job *batch.Job, nodeName string, config *provisionConfig, action string) bool {
-	if err := k8sutil.RunReplaceableJob(c.context.Clientset, job, false); err != nil {
-		if !kerrors.IsAlreadyExists(err) {
-			// we failed to create job, update the orchestration status for this node
-			message := fmt.Sprintf("failed to create %q job for node %q. %v", action, nodeName, err)
-			c.handleOrchestrationFailure(config, nodeName, message)
-			return false
-		}
-
-		// the job is already in progress so we will let it run to completion
-	}
-
-	logger.Infof("osd %s job started for node %s", action, nodeName)
-	return true
-}
-
-func (c *Cluster) startOSDDaemonsOnPVC(pvcName string, config *provisionConfig, configMap *v1.ConfigMap, status *OrchestrationStatus) {
-	ctx := context.TODO()
-	osds := status.OSDs
-	logger.Infof("starting %d osd daemons on pvc %s", len(osds), pvcName)
 	osdProps, err := c.getOSDPropsForPVC(pvcName)
 	if err != nil {
-		config.addError(fmt.Sprintf("%v", err))
-		return
+		return errors.Wrapf(err, "failed to get properties for %s", osdLongName)
 	}
 
-	// start osds
-	for _, osd := range osds {
-		logger.Debugf("start osd %v", osd)
-
-		dp, err := c.makeDeployment(osdProps, osd, config)
-		if err != nil {
-			errMsg := fmt.Sprintf("failed to create deployment for pvc %q. %v", osdProps.crushHostname, err)
-			config.addError(errMsg)
-			continue
-		}
-
-		// Set the deployment hash as an annotation
-		err = patch.DefaultAnnotator.SetLastAppliedAnnotation(dp)
-		if err != nil {
-			errMsg := fmt.Sprintf("failed to set annotation for deployment %q. %v", dp.Name, err)
-			config.addError(errMsg)
-			continue
-		}
-
-		_, createErr := c.context.Clientset.AppsV1().Deployments(c.clusterInfo.Namespace).Create(ctx, dp, metav1.CreateOptions{})
-		if createErr != nil {
-			if kerrors.IsAlreadyExists(createErr) {
-				logger.Infof("deployment for osd %d already exists. updating if needed", osd.ID)
-				if err = updateDeploymentAndWait(c.context, c.clusterInfo, dp, opconfig.OsdType, strconv.Itoa(osd.ID), c.spec.SkipUpgradeChecks, c.spec.ContinueUpgradeAfterChecksEvenIfNotHealthy); err != nil {
-					logger.Errorf("failed to update osd deployment %d. %v", osd.ID, err)
-				}
-			} else {
-				// we failed to create job, update the orchestration status for this pvc
-				logger.Warningf("failed to create osd deployment for pvc %q, osd %v. %v", osdProps.pvc.ClaimName, osd, createErr)
-				continue
-			}
-		}
+	// TODO: why was there not a Progressing condition added here before?
+	// opconfig.ConditionExport(c.context, c.clusterInfo.NamespacedName(), cephv1.ConditionProgressing, v1.ConditionTrue, "ClusterProgressing", fmt.Sprintf("Processing node %s osd %d", nodeName, osd.ID))
```

Comment:
@travisn @leseb ?

Reply:
The progressing condition is fairly new, it must have been missed for the OSDs on PVCs